integration/common/openlineage/common/provider/dbt.py

```diff
@@ -175,13 +201,15 @@ def parse(self) -> DbtEvents:
             if name.startswith('model.') or name.startswith('test.'):
                 nodes[name] = node
 
-        if run_result['args']['which'] == 'run':
-            return self.parse_run(manifest, run_result, catalog, nodes)
-        elif run_result['args']['which'] == 'test':
-            return self.parse_test(manifest, run_result, catalog, nodes)
+        context = DbtRunContext(manifest, run_result, catalog)
+
+        if self.command in ['run', 'build']:
+            return self.parse_execution(context, nodes)
+        elif self.command == 'test':
+            return self.parse_test(context, nodes)
```

Comment:
Minor issue, but I would keep the `parse_execution` method ignorant of the `build` semantics and move that logic here so the workflow of each command is handled in one central place. E.g., something like

```python
events = DbtEvents()
if self.command in ['run', 'build']:
    events.add(self.parse_execution(context, nodes))
if self.command in ['test', 'build']:
    events.add(self.parse_test(context, nodes))
return events
```

Reply:
Done.